### ISSUE TITLE
Is it possible to add japanese pokemon names?

### ISSUE THREAD

**Issue by saboyutaka** [LazoCoder/Pokemon-Terminal#103]
Pokemon terminal is awesome 👍 👍 👍  
But  I think it's better for non-english user if you can choose your country's pokemon name
 e.x Bulbasaur is フシギダネ in japanese.

here is a list for pokemon names in many languages. (It's in Japanese though )
https://wiki.ポケモン.com/wiki/ポケモンの外国語名一覧

If it's possible. I'll code it.

**Comment by charlesmilette** [comment]
Is Japanese text input even supported on terminal emulators?
The Japanese characters being wider than a regular monospaced font, won't
that mess up the terminal?

Le jeu. 27 juil. 2017 00:23, Yutaka Tachibana <notifications@github.com> a
écrit :

**Comment by saboyutaka** [comment]
> Is Japanese text input even supported on terminal emulators?
Yeah, I think so.

Once I edit `Data/pokemon.txt` from english to japanese name, It works.

![pokemon](https://i.gyazo.com/0f73427f708e743a5417372e2efe2ef0.gif)

**Comment by charlesmilette** [comment]
Yeah, it's just a regular string search. What I was wondering is if you
could even enter Japanese characters in the terminal correctly. It messes
up enough when I try a simple Unicode character.

Le jeu. 27 juil. 2017 00:46, Yutaka Tachibana <notifications@github.com> a
écrit :

**Comment by saboyutaka** [comment]
UTF-8 includes Japanese characters(Katakana) which is used pokemon names.

https://www.key-shortcut.com/en/writing-systems/ひらがな-japanese/

**Comment by cclauss** [comment]
To do this, we would need a new version of https://github.com/LazoCoder/Pokemon-Terminal/blob/master/Data/pokemon.txt perhaps named pokemon-jp.txt that **added a second column** which was the Japanese name of each character.  Those names must have no whitespaces!  With that file, we could do some experimentation to see how hard it would be to deliver this feature.

**Comment by Squirrels** [comment]
I was going to add support for japanese, getting the names from [Sindresorhus' Pokemon Name List](https://github.com/sindresorhus/pokemon), which contains japanese, russian, among other languages, when I started wondering if we could add support for all of those languages at once.

The problem would be that the pokemon.txt file would keep growing, and may become too messy. I was thinking converting it to a JSON file, where adding more information to each Pokemon would be easier. Opinions, @cclauss @LazoCoder @sylveon ?

**Comment by cclauss** [comment]
I would still vote no to json but I might be in the minority. https://github.com/LazoCoder/Pokemon-Terminal/pull/96

**Comment by sylveon** [comment]
What about csv?

**Comment by samosaara** [comment]
I wanted to change the db into JSON ages ago, but since it were basically the pokemon name and the file, it was just going to overly complicate it, but now we have, the lightness index, the pokemon types, and the pokemon regions are all implicit, support for localized pokemon names might just be the nail in the coffin for the change, the problem is, @LazoCoder is, well, occupied? And we already have tons of conflicting changes we will have to sort out, I would love to change it, just don't think the timing is right. We could even finally implement pokemon named tuples (right @cclauss ?) to easily marshal and unmarshal them, not to mention the almost assured speed increase

**Comment by samosaara** [comment]
Some work has been done in #96 that I think can be salvaged

**Comment by Squirrels** [comment]
Yeah, I'm planning on using that as a base, updating with the new pokemon and information as soon as I'm done with #112 and Mac OS testing for #94 (our current master).

**Comment by cclauss** [comment]
OK.  No more flatfile!
